plugins/mkt-queue-sync/src/classes/Synchronizer.php

```diff
@@ -4,252 +4,472 @@
 
 namespace MikrotikQueueSync;
 
-use Ubnt\UcrmPluginSdk\Security\PermissionNames;
 use Ubnt\UcrmPluginSdk\Service\UcrmApi;
 use Ubnt\UcrmPluginSdk\Service\UcrmOptionsManager;
 use Ubnt\UcrmPluginSdk\Service\PluginConfigManager;
 use Ubnt\UcrmPluginSdk\Service\PluginLogManager;
+use Ubnt\UcrmPluginSdk\Exception\ConfigurationException;
 
 require __DIR__ . '/../vendor/autoload.php';
 
-$GLOBALS['sumDownloadLimitAt'] = 0;
-$GLOBALS['sumUploadLimitAt'] = 0;
-$GLOBALS['sumDownloadVendido'] = 0;
-$GLOBALS['sumUploadVendido'] = 0;
-
 class Synchronizer
 {
-    public function __construct() {
-    }
+	private const COMMENT_SIGNATURE = 'ucrm_mktsync_';
+	
+	/**
+     * @var UcrmApi
+     */
+    private $ucrmApi;
+    
+	/**
+     * @var Logger
+     */
+    private $logger;
+	
+	/**
+     * @var pluginConfigManager
+     */
+    private $pluginConfigManager;
+    
+	/**
+     * @var UnmsApi
+     */
+    private $unmsApi; //Enable only for UNMS v1
+
+    /**
+     * @var RouterOsApi
+     */
+    private $routerOsApi;
+	
+	/**
+     * @var SyncAddressList
+     */
+    //private $syncAddressList;
+
+	/**
+     * @var RemoveEnded
+     */
+    //private $removeEnded;
 
+    public function __construct()
+    {
+        $this->ucrmApi = UcrmApi::create();
+		$this->logger = PluginLogManager::create();
+		$this->pluginConfigManager = PluginConfigManager::create();
+        if (!(substr(($this->ucrmApi->get('version')['version']),0,-strlen($this->ucrmApi->get('version')['version'])+1) == 2)){
```

Comment:
You can check for UNMS integrated version by checking for `unmsLocalUrl` in `UcrmOptions`, take a look at https://github.com/Ubiquiti-App/UCRM-Plugin-SDK/pull/14 (available since 1.0.0-beta.2)

For example:
```php
if ((new UcrmOptionsManager())->loadOptions()->unmsLocalUrl !== null) {
    // UNMS v1
} else {
    // UCRM v2
}

Reply:
Hi Keksa.... Ahhhh, ok! You're right. I'll improve it in my next release. Thanks. Have a nice day